go.mod

```
@@ -292,6 +292,9 @@ require (
 )
 
 replace (
+	// Test new functionality from gitops-engine
+	github.com/argoproj/gitops-engine => github.com/andrii-korotkov-verkada/gitops-engine v0.0.0-20240707213928-5b2dacf3ee42
```

Since it is not open in draft, commenting this so it isn't merge.

```suggestion
	github.com/argoproj/gitops-engine => github.com/andrii-korotkov-verkada/gitops-engine v0.0.0-20240707213928-5b2dacf3ee42
```